src/Pos/Generator/Block/Logic.hs

```diff
@@ -36,17 +42,18 @@ import           Pos.Util.Util             (maybeThrow, _neHead)
 -- valid with respect to the global state right before this function
 -- call.
 genBlocks ::
-       MonadBlockGen ctx m
+       (MonadBlockGen ctx m, RandomGen g)
     => BlockGenParams
+    -> g
```

Comment:
Export it with `RandT`, not with `g` as a simple function argument. The reason is that if you're abstracting over `g`, you can't assume that it's read-only, it needs to be passed in a `StateT`-like manner. Therefore you'll need to return `(OldestFirst [] (Blund SscGodTossing), g)`. But it's not as convenient as `RandT` because then the caller doesn't have a choice between `evalRandT` and `runRandT`).

For instance, in my code I will need to call `genBlocks` in the `RandT` monad directly, and I don't want unnecessary wrapping/unwrapping.